compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -1574,6 +1574,12 @@ public void visit(BLangThrow throwNode) {
 
     public void visit(BLangPanic panicNode) {
         this.checkStatementExecutionValidity(panicNode);
+
+        // Check whether the enclosed function's return type is 'nil'
+        if (this.env.enclInvokable.returnTypeNode.type.tag == TypeTags.NIL) {
+            dlog.warning(this.env.enclInvokable.returnTypeNode.pos, DiagnosticCode.FUNCTION_DOES_NOT_RETURN);
```

Comment:
Is this only applicable for functions returning `()`? I was under the impression that this applies to any function in which a panic is used. e.g.,
```ballerina
function foo() returns string|never {
    // some code
    
    if (true) {
        panic error("ERROR");
    }

    // more code

    return "foo";
}
```

Reply:
@hasithaa thoughts?

Reply:
Removed this part as discussed